js/core/component.js

```diff
@@ -275,84 +276,78 @@ var Component = Class.inherit({
-            actionFunc;
+        let action;
+        let eventName;
+        let actionFunc;
 
-        var result = function() {
+        let result = (...args) => {
             if(!eventName) {
                 config = config || {};
 
-                if(typeof optionName !== "string") {
-                    throw errors.Error("E0008");
+                if(typeof optionName !== 'string') {
+                    throw errors.Error('E0008');
                 }
 
-                if(optionName.indexOf("on") === 0) {
-                    eventName = that._getEventName(optionName);
+                if(optionName.indexOf('on') === 0) {
+                    eventName = getEventName(optionName);
                 }
                 ///#DEBUG
-                if(optionName.indexOf("on") !== 0) {
-                    throw Error("The '" + optionName + "' option name should start with 'on' prefix");
+                if(optionName.indexOf('on') !== 0) {
+                    throw Error(`The '${optionName}' option name should start with 'on' prefix`);
                 }
                 ///#ENDDEBUG
 
-                actionFunc = that.option(optionName);
+                actionFunc = this.option(optionName);
             }
 
-            if(!action && !actionFunc && !config.beforeExecute && !config.afterExecute && !that._eventsStrategy.hasEvent(eventName)) {
+            if(!action && !actionFunc && !config.beforeExecute && !config.afterExecute && !this._eventsStrategy.hasEvent(eventName)) {
                 return;
             }
 
             if(!action) {
-                var beforeExecute = config.beforeExecute;
-                config.beforeExecute = function(args) {
-                    beforeExecute && beforeExecute.apply(that, arguments);
-                    that._eventsStrategy.fireEvent(eventName, args.args);
+                const beforeExecute = config.beforeExecute;
+                config.beforeExecute = (...props) => {
+                    beforeExecute && beforeExecute.apply(this, props);
+                    this._eventsStrategy.fireEvent(eventName, props[0].args);
                 };
-                action = that._createAction(actionFunc, config);
+                action = this._createAction(actionFunc, config);
             }
 
             if(Config().wrapActionsBeforeExecute) {
-                var beforeActionExecute = that.option("beforeActionExecute") || noop;
-                var wrappedAction = beforeActionExecute(that, action, config) || action;
-                return wrappedAction.apply(that, arguments);
+                const beforeActionExecute = this.option('beforeActionExecute') || noop;
+                const wrappedAction = beforeActionExecute(this, action, config) || action;
+                return wrappedAction.apply(this, args);
             }
 
-            return action.apply(that, arguments);
+            return action.apply(this, args);
         };
 
         if(!Config().wrapActionsBeforeExecute) {
```

Comment:
```suggestion
         if(Config().wrapActionsBeforeExecute || !this.option('onActionCreated')) {
           return result;
         }

         return this.option('onActionCreated')(this, result, config) || result;
```

Since there are multiple `return` statements here already

Reply:
I refactored it. Check please